packages/application-components/src/components/custom-views/custom-view-loader/custom-view-loader.tsx

```diff
@@ -0,0 +1,154 @@
+  } catch {
+    // Trying to access the contentWindow of a cross-origin iFrame will throw an error.
+    // We are not supposed to even get here because the iFrame must use
+    // a URL from our very same domain (the custom view is proxied through our http-proxy service).
+    return false;
+  }
+};
+
+const CustomPanelIframe = styled.iframe`
+  height: 100%;
+  width: 100%;
+  border: none;
+`;
+
+function CustomViewLoader(props: TCustomViewLoaderProps) {
+  const iFrameElementRef = useRef<HTMLIFrameElement>(null);
+  const appContext = useApplicationContext();
+  const iFrameCommunicationChannel = useRef(new MessageChannel());
+  const showNotification = useShowNotification();
+  const intl = useIntl();
+  const panelSize = (props.customView.typeConfig?.size?.toLocaleLowerCase() ||
+    'large') as Lowercase<'SMALL' | 'LARGE'>;
+
+  const messageFromIFrameHandler = useCallback((event: MessageEvent) => {
+    if (event.data.origin === window.location.origin) {
+      console.log('message received from iframe port: ', event);
+    }
+  }, []);
+
+  // onLoad handler is called from the iFrame even where the URL is not valid
+  // (blocked by CORS, 404, etc.) so we need to make sure the iFrame is ready
+  const onLoadSuccessHandler = useCallback(() => {
+    // Show error and block if the iFrame is not ready
+    // (error loading it)
+    if (!isIframeReady(iFrameElementRef.current!)) {
+      showNotification({
+        domain: DOMAINS.PAGE,
+        kind: NOTIFICATION_KINDS_PAGE.error,
+        text: intl.formatMessage(messages.loadError),
+      });
+      return;
+    }
+
+    // Listen for messages from the iFrame
+    iFrameCommunicationChannel.current.port1.onmessage =
+      messageFromIFrameHandler;
+
+    // Transfer port2 to the iFrame so it can send messages back privately
+    iFrameElementRef.current?.contentWindow?.postMessage(
+      CUSTOM_VIEWS_EVENTS_NAMES.CUSTOM_VIEW_BOOTSTRAP,
+      window.location.href,
+      [iFrameCommunicationChannel.current.port2]
+    );
+
+    // Send the initialization message to the iFrame
+    iFrameCommunicationChannel.current.port1.postMessage({
+      source: CUSTOM_VIEWS_EVENTS_META.SOURCE,
+      destination: `${CUSTOM_VIEWS_EVENTS_META.DESTINATION_PREFIX}${props.customView.id}`,
+      eventName: CUSTOM_VIEWS_EVENTS_NAMES.CUSTOM_VIEW_INITIALIZATION,
+      eventData: {
+        context: {
+          userLocale: appContext.user?.locale,
+          dataLocale: appContext.dataLocale,
+          projectKey: appContext.project?.key,
+        },
+      },
+    } as TCustomViewIframeMessage);
+
+    // We want the effect to run only once so we don't need the dependencies array.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // Currently we only support custom panels
+  if (props.customView.type !== 'CustomPanel') {
+    reportErrorToSentry(
+      new Error(
+        `CustomViewLoader: Provided Custom View has an unsupported type: ${props.customView.type}`
```

Comment:
Nit: I'm thinking maybe it might be useful to add in the error message what is the only supported type at the moment. 
Something like: `CustomViewLoader: The Custom View has an unsupported type: ${props.customView.type. Presently, the sole supported type is 'CustomPanel'.`

Reply:
Thanks for the suggestion 👍 
I updated the message here: 744c59951b328d9dac65dd2b09fcd11cc814e981

Bear in mind this is a very unlikely error to happen and it's more for internal debug that customers facing ([explanation](https://github.com/commercetools/merchant-center-application-kit/pull/3151#discussion_r1281688722)).